storage/remote/otlptranslator/prometheusremotewrite/helper.go

```diff
@@ -122,12 +122,13 @@ func createAttributes(resource pcommon.Resource, attributes pcommon.Map, setting
 	serviceName, haveServiceName := resourceAttrs.Get(conventions.AttributeServiceName)
 	instance, haveInstanceID := resourceAttrs.Get(conventions.AttributeServiceInstanceID)
 
-	promotedAttrs := make([]prompb.Label, 0, len(settings.PromoteResourceAttributes))
-	for _, name := range settings.PromoteResourceAttributes {
-		if value, exists := resourceAttrs.Get(name); exists {
+	promotedAttrs := make([]prompb.Label, 0, resourceAttrs.Len())
+	resourceAttrs.Range(func(name string, value pcommon.Value) bool {
+		if settings.ResourceAttributesSetting.shouldPromote(name) {
 			promotedAttrs = append(promotedAttrs, prompb.Label{Name: name, Value: value.AsString()})
 		}
-	}
+		return true
+	})
 	sort.Stable(ByLabelName(promotedAttrs))
```

Comment:
If we are going to abstract `settings.PromoteResourceAttributes`, let's do it fully and avoid a leaky abstraction:

```suggestion
promotedAttrs := settings.PromoteResourceAttributes.promotedAttributes(resourceAttrs)
```